include/k4ainternal/rwlock.h

```diff
@@ -0,0 +1,48 @@
+/** \file rwlock.h
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ * Kinect For Azure SDK.
+ */
+
+#ifndef RWLOCK_H
+#define RWLOCK_H
+
+#include <k4a/k4atypes.h>
+
+#include <stddef.h>
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <pthread.h>
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct
```

Comment:
This should be hidden in the c file

Reply:
Since the structure is different sizes on Windows and Linux, and I'd prefer to keep the lock allocation free, I think this is the cleanest we can do.